platforms/jvm/language-jvm/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -113,117 +93,149 @@ public CompileOptions(ObjectFactory objectFactory) {
 
     /**
-     * Sets whether to fail the build when compilation fails. Defaults to {@code true}.
+     * TODO: Add deprecation warning
+     *
+     * @since 8.9
      */
-    public void setFailOnError(boolean failOnError) {
-        this.failOnError = failOnError;
+    @Incubating
+    @ReplacedBy("failOnError")
+    public Property<Boolean> getIsFailOnError() {
+        return getFailOnError();
     }
 
     /**
      * Tells whether to produce verbose output. Defaults to {@code false}.
+     *
+     * @since 8.9
      */
     @Console
-    public boolean isVerbose() {
-        return verbose;
-    }
+    @Incubating
+    @ReplacesEagerProperty(originalType = boolean.class)
+    public abstract Property<Boolean> getVerbose();
 
     /**
-     * Sets whether to produce verbose output. Defaults to {@code false}.
+     * TODO: Add deprecation warning
+     *
+     * @since 8.9
      */
-    public void setVerbose(boolean verbose) {
-        this.verbose = verbose;
+    @Incubating
+    @ReplacedBy("verbose")
+    public Property<Boolean> getIsVerbose() {
+        return getVerbose();
     }
 
     /**
      * Tells whether to log the files to be compiled. Defaults to {@code false}.
+     *
+     * @since 8.9
      */
     @Console
-    public boolean isListFiles() {
-        return listFiles;
-    }
+    @Incubating
+    @ReplacesEagerProperty(originalType = boolean.class)
+    public abstract Property<Boolean> getListFiles();
 
     /**
-     * Sets whether to log the files to be compiled. Defaults to {@code false}.
+     * TODO: Add deprecation warning
+     *
+     * @since 8.9
      */
-    public void setListFiles(boolean listFiles) {
-        this.listFiles = listFiles;
+    @Incubating
+    @ReplacedBy("listFiles")
+    public Property<Boolean> getIsListFiles() {
+        return getListFiles();
     }
 
     /**
      * Tells whether to log details of usage of deprecated members or classes. Defaults to {@code false}.
+     *
+     * @since 8.9
      */
     @Console
-    public boolean isDeprecation() {
-        return deprecation;
-    }
+    @Incubating
+    @ReplacesEagerProperty(originalType = boolean.class)
+    public abstract Property<Boolean> getDeprecation();
 
     /**
-     * Sets whether to log details of usage of deprecated members or classes. Defaults to {@code false}.
+     * TODO: Add deprecation warning
+     *
+     * @since 8.9
      */
-    public void setDeprecation(boolean deprecation) {
-        this.deprecation = deprecation;
+    @Incubating
+    @ReplacedBy("deprecation")
+    public Property<Boolean> getIsDeprecation() {
+        return getDeprecation();
     }
 
     /**
      * Tells whether to log warning messages. The default is {@code true}.
+     *
+     * @since 8.9
      */
     @Console
-    public boolean isWarnings() {
-        return warnings;
-    }
+    @Incubating
+    @ReplacesEagerProperty(originalType = boolean.class)
+    public abstract Property<Boolean> getWarnings();
```

Comment:
❓ Why mark these methods as `@Incubating`?

Reply:
👍 There's a bunch of other places, too. Maybe this was added before we had automatic acceptance of binary compatibility changes..?

Reply:
I've opened https://github.com/gradle/gradle/issues/29811, we need better handling for `boolean` properties for binary compatibilty changes